imap_processing/hi/l1c/hi_l1c.py

```diff
@@ -0,0 +1,221 @@
+
+from imap_processing import imap_module_directory
+from imap_processing.cdf.cdf_attribute_manager import CdfAttributeManager
+from imap_processing.cdf.utils import load_cdf
+
+logger = logging.getLogger(__name__)
+
+
+def hi_l1c(dependencies: list, data_version: str):
+    """
+    High level IMAP-Hi l1c processing function.
+
+    This function will be expanded once the l1c processing is better defined. It
+    will need to add inputs such as Ephemerides, Goodtimes inputs, and
+    instrument status summary and will output a Pointing Set CDF as well as a
+    Goodtimes list (CDF?).
+
+    Parameters
+    ----------
+    dependencies : list
+        Input dependencies needed for l1c processing.
+
+    data_version : str
+        Data version to write to CDF files and the Data_version CDF attribute.
+        Should be in the format Vxxx
+
+    Returns
+    -------
+    processed_data : xarray.Dataset
+        Processed xarray dataset
+    """
+    logger.info("Running Hi l1c processing")
+
+    # TODO: I am not sure what the input for Goodtimes will be so for now,
+    #    If the input is a CDF, do pset processing
+    if len(dependencies) == 1 and str(dependencies[0]).endswith("cdf"):
+        l1c_dataset = generate_pset_dataset(dependencies[0])
+    else:
+        raise NotImplementedError(
+            "Input dependencies not recognized for l1c pset processing."
+        )
+
+    # TODO: revisit this
+    l1c_dataset.attrs["Data_version"] = data_version
+    return l1c_dataset
+
+
+def generate_pset_dataset(l1b_de_path: Union[Path, str]) -> xr.Dataset:
+    """
+    Process IMAP-Hi l1b product into a l1c pset xarray dataset.
+
+    Parameters
+    ----------
+    l1b_de_path : Union[Path, str]
+        Location of IMAP-Hi l1b de product.
+
+    Returns
+    -------
+    xarray.Dataset ready to be written to CDF.
+    """
+    de_dataset = load_cdf(l1b_de_path)
+    sensor_str = de_dataset.attrs["Logical_source"].split("_")[-1].split("-")[0]
+    n_esa_step = de_dataset.esa_step.data.size
+    pset_dataset = allocate_pset_dataset(n_esa_step, sensor_str)
+    # TODO: Stored epoch value needs to be consistent across ENA instruments.
+    #    SPDF says this should be the center of the time bin, but instrument
+    #    teams may disagree.
+    pset_dataset.epoch.data[0] = de_dataset.epoch.data[0]
+
+    return pset_dataset
+
+
+def allocate_pset_dataset(n_esa_steps: int, sensor_str: str):
+    """
+    Allocate an empty xarray.Dataset.
+
+    Parameters
+    ----------
+    n_esa_steps : int
+        Number of Electrostatic Analyzer steps to allocate
+    sensor_str : str
+        '45sensor' or '90sensor'
+
+    Returns
+    -------
+    xarray.Dataset
+        Empty xarray.Dataset ready to be filled with data
+    """
+    cdf_manager = CdfAttributeManager(imap_module_directory / "cdf" / "config")
+    cdf_manager.load_global_attributes("imap_hi_global_cdf_attrs.yaml")
+    cdf_manager.load_variable_attributes("imap_hi_variable_attrs.yaml")
```

Comment:
This is ok for now. After SIT-3, we will need to update this to this. I believe this does minor validation and sounds like we may expand this afterwards.
```
    # create the attribute manager for this data level
    attr_mgr = ImapCdfAttributes()
    attr_mgr.add_instrument_global_attrs(instrument="lo")
    attr_mgr.add_instrument_variable_attrs(instrument="lo", level="l1a")
    attr_mgr.add_global_attribute("Data_version", data_version)
```